v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -123,14 +123,38 @@ function formatSingleError(
   if (!message.includes(error.message)) {
     message = error.message;
   }
+
+  // We remove the node:assert and node:test error codes
   message = message
     .replace(" [ERR_ASSERTION]", "")
     .replace(" [ERR_TEST_FAILURE]", "");
 
   const diff = getErrorDiff(error);
 
   if (diff !== undefined) {
-    message = message.match(/^(AssertionError:\s.*)\:/)?.[1] ?? message;
+    // If we got a diff, we try to remove any diff from the message, which can
+    // have different formats.
+
+    // Format 1: A line starting with "+ actual - expected".
+    let match = message.match(/^(.*?)\n\s*?\+\sactual\s-\sexpected/s);
+
+    // Format 2: A line starting with "{error.actual} !==".
+    if (match === null && "actual" in error) {
+      match = message.match(/^(.*?)\n\s*?(?:.*?) !==/s);
+    }
+
+    // Format 3: AssertionError: {message}:
+    match = match ?? message.match(/^(AssertionError:\s.*)\:/);
```

Comment:
I'm thinking that maybe we could remove this altogether because we're not really sure what we're removing here, right? In 1 and 2, we're removing known diff formats, which should cover most of the cases. And if we new diff formats are added, we can always add them.

Reply:
That's a good point!

Reply:
OK, I'm going to remove it then 💪